portfolio-website/app/components/Navigation/Navigation.tsx

```diff
@@ -0,0 +1,84 @@
+'use client';
+import Image from "next/image";
+import { useEffect } from "react";
+
+export const Navigation = ({ currentSection }: { currentSection: string }) => {
+    function toggleNavigation() {
+        const burgerBtn = document.querySelector('.burger-menu-button') as HTMLButtonElement;
+        const navigation = document.querySelector('.side-nav-mid') as HTMLUListElement;
+
+        burgerBtn.classList.toggle('close-nav');
+        navigation.classList.toggle('active-mobile-nav');
+    }
+
+    function handleNavLinkClick() {
+        const burgerBtn = document.querySelector('.burger-menu-button') as HTMLButtonElement;
+        const navigation = document.querySelector('.side-nav-mid') as HTMLUListElement;
+        if (navigation.classList.contains('active-mobile-nav')) {
+            navigation.classList.remove('active-mobile-nav');
+            burgerBtn.classList.remove('close-nav');
+        }
+    }
+
+    useEffect(() => {
+        const handleOutsideClick = (ev: MouseEvent) => {
+            const burgerBtn = document.querySelector('.burger-menu-button') as HTMLButtonElement;
```

Comment:
this is now resolved, and a lot more elegant solution is present, but since i haven't added the rest of the components, i cannot assure you that it still works as expected